### PR TITLE
Drain controllers: narrow down interfaces

### PR DESCRIPTION
**Pull request by fossedihelm**
### What this PR does
There is no need to hold informers, narrowing down
the interface can decrease the probability of making
a mistake. It also simplifies the controller and unit
tests.

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-controller/watch/drain/evacuation/evacuation.go [hunk 1]
```diff
@@ -72,7 +73,11 @@ func NewEvacuationController(
 		clusterConfig:         clusterConfig,
 	}
 
-	_, err := c.vmiInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+	c.hasSynced = func() bool {
+		return vmiInformer.HasSynced() && vmiPodInformer.HasSynced() && migrationInformer.HasSynced() && nodeInformer.HasSynced() && vmiPodInformer.HasSynced()
```
Comment (xpivarc) [inline pkg/virt-controller/watch/drain/evacuation/evacuation.go:77]: `vmiPodInformer` is here twice?

**Review by xpivarc — COMMENTED** [review]
/approve

File: pkg/virt-controller/watch/drain/evacuation/evacuation.go (hunk 1 above)
Comment (fossedihelm) [reply pkg/virt-controller/watch/drain/evacuation/evacuation.go]: too aggressive Ctrl+V

Comment (fossedihelm) [reply pkg/virt-controller/watch/drain/evacuation/evacuation.go]: Thanks

**Review by 0xFelix — APPROVED** [review]
/lgtm

**Pull request merged**